tests/Checker/Preparations/Force_Single_Plugin_Preparation_Tests.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Tests for the Force_Single_Plugin_Preparation class.
+ *
+ * @package plugin-check
+ */
+
+namespace Checker\Preparations;
+
+use WordPress\Plugin_Check\Checker\Preparations\Force_Single_Plugin_Preparation;
+use WP_UnitTestCase;
+use Exception;
+
+class Force_Single_Plugin_Preparation_Tests extends WP_UnitTestCase {
+
+	public function test_prepare_plugin_exists() {
+
+		$preparation = new Force_Single_Plugin_Preparation( 'akismet/akismet.php' );
+		$message     = '';
+
+		try {
+			$preparation->prepare();
+		} catch ( Exception $e ) {
+			$message = $e->getMessage();
+		}
+
+		$this->assertEquals( 'The plugin akismet/akismet.php does not exists', $message );
+	}
+
+	public function test_prepare() {
+
+		$preparation = new Force_Single_Plugin_Preparation( 'plugin-check/plugin-check.php' );
+
+		$plugins = array(
+			'akismet/akismet.php',
+			'plugin-check/plugin-check.php',
+			'wp-reset/wp-reset.php',
+		);
+
+		update_option( 'active_plugins', $plugins );
+
+		$cleanup = $preparation->prepare();
+
+		$active_plugins = get_option( 'active_plugins' );
+
+		$cleanup();
+
+		$this->assertIsCallable( $cleanup );
+
+		$this->assertContains( 'plugin-check/plugin-check.php', $active_plugins );
+
+		$this->assertEquals(
+			array(
+				'plugin-check/plugin-check.php',
+				'plugin-check/plugin-check.php',
+			),
+			$active_plugins
+		);
```

Comment:
Can you add another assertion here to ensure the filters have been cleaned up again? Basically call `get_option( 'active_plugins' )` again and assert that it has the original 3 plugin basenames in it.

Reply:
@felixarntz I have added that test, but we don't need to set the `active_plugins` option for this. We can directly pass the plugin files that are not added while `Force_Single_Plugin_Preparation` instantiating so it will return the actual files array we send - You can check code for that assertion here - https://github.com/10up/plugin-check/pull/68/files#diff-4a0736f42c4d3624b9741527c78372d58c249c9fa517ad3df1c15cb78307627cR87

Reply:
That assertion you're referring to is not the same though. That assertion tests whether the `filter_active_plugins()` method works correctly.

I think we need another assertion in this method here, to assert that _after_ calling the `$cleanup()` method, the return value from `get_option( 'active_plugins' )` is the same as what we originally specified - not because of how `filter_active_plugins()` works, but because the filter callback should have been removed entirely.

Reply:
@felixarntz We have one issue/blocker with this situation: the `active_plugins` setting, defined (see below) in the `bootstrap.php` that always overrides the `active_plugins` option which we set from the individual tests functions. And it is always returning the `plugin-check/plugin-check.php` file as the single active plugin. I'm finding out the solutions to it, maybe via something like `tests_add_filter`, but at the moment I'm not sure how this will be fixed.

```
$GLOBALS['wp_tests_options'] = array(
  'active_plugins' => array( basename( TESTS_PLUGIN_DIR ) . '/plugin-check.php' ),
);
```

Reply:
@vishalkakadiya Looking at the WP core test suite code should provide some clarity on how we can solve this. I think we need to simply remove that filter on top of this test method?

Maybe try this:
```php
remove_filter( 'pre_option_active_plugins', 'wp_tests_options' );
```

Reply:
@felixarntz Yeah, it is working fine with this solution. Thank you! 🙌 

@jjgrainger Thank you for committing that code. 🙇